app/models/user.rb

```diff
@@ -0,0 +1,31 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  property :id,              Serial
+  property :email,           String, unique: true
+  property :password_digest, Text
+  property :name,            String
+  property :username,        String, unique: true
+
+  #has n, :peeps, through: Resource
+
+  attr_reader :email
+  attr_reader :password
+  attr_reader :name
+  attr_reader :username
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.login(username, password)
+    user = first(username: username)
+    if user && BCrypt::Password.new(user.password_digest) == password
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.